enhancements/monitoring/monitoring-stack-operator.md

```diff
@@ -0,0 +1,490 @@
+* With single (or HA) Prometheus, this stack will not scale. In order to scale we would need to deploy Observatorium which is [explained](#full-forwarding-mode-with-local-external-or-hosted-observatorium) in a previous alternative.
+* The same topology will be reproducible with the proposed alternative by specifying multiple namespaces in the `MonitoringStack` resource.
+
+### Shared Thanos Querier
+
+Pros:
+
+* Less resource overhead
+* Global view out of the box
+
+Cons:
+
+* This approach makes a `MonitoringStack` coupled to a centralized component. In order for individual teams to query their own metrics, they will need to query all prometheus instances.
+* Currently Thanos Query does not have good QoS mechanisms for independent query targets. Intensive queries from one team could therefore impact querying capabilities of other teams.
+* Since we want to build a community around the project, this approach makes the stack overall less composable and less reusable. It assumes that `MonitoringStack` owners are fine with exposing their metrics to each other. This might be the case for Managed Services, but it does not necessarily apply in the general sense.
+* In case a central view is needed, an additional Thanos Query instance can still be deployed independently.
+
+### Use the existing monitoring stack managed by Cluster Monitoring Operator
+
+This alternative was explored and was designated as not viable in OSD clusters. The Platform Monitoring stack is completely controlled by OSD SREs, and User Workload Monitoring is used by end customers. Trying to integrate managed services into this stack would be challenging and risky. We haven’t been able to find a good solution using this approach.
+
+### Deploy a third Prometheus instance through Cluster Monitoring Operator
+
+While this is a viable alternative, deploying a third instance through CMO will merely be a stop-gap solution that can be challenging to reuse for upcoming use-cases, such as hypershift. In addition, coupling managed service monitoring to OCP releases will reduce delivery velocity, thereby inhibiting agile software development and iterative value delivery.
+An OLM-based stack, on the other hand, allows us to release as often as possible and deliver value to managed service owners faster and in incremental batches.
+
+Furthermore, CMO in the current state already takes on a lot of responsibility. It defines and manages two stacks, Platform Monitoring and User Workload Monitoring. Expanding the scope of CMO further would make it harder to maintain in the long run.
+
+### Upgrade / Downgrade Strategy
+
+New versions of the managed service monitoring stack will be published in OLM. Upgrades and downgrades will be performed by OLM itself.
+
+### Graduation Criteria
+
+TBD
+
+#### Dev Preview -> Tech Preview
+
+TBD
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+TBD
+
+### Version Skew Strategy
```

Comment:
How do we plan to address API version drifts between the built in CMO vs the MonitoringStack Operator? Does the MSO simply not bring any CRD's along?

Reply:
This has to be solved with OLM help. The new scoped/descoped semantics should help.

Reply:
Still, something to clarify